tensor_genn/algorithms/relu_ann.py

```diff
@@ -118,36 +150,47 @@ def convert(self, tf_model):
         cs_init = {"magnitude":10.0}
 
         # Fetch augmented weight matrices
-        g_weights, n_units = self.create_weight_matrices(tf_model)
-        gw_inds = [np.nonzero(gw) for gw in g_weights]
-        gw_vals = [g_weights[i][gw_inds[i]].reshape(-1) for i in range(len(g_weights))]
+        gw_inds, gw_vals, n_units, relevant_layers = self.create_weight_matrices(tf_model)
 
         # Define model and populations
         self.g_model = genn_model.GeNNModel("float","g_model")
         self.neuron_pops = []
         self.syn_pops = []
 
-        for i in range(1,n_layers):
+        for i in range(1,len(relevant_layers)+1):
             if i == 1:
                 # Presynaptic neuron
                 self.neuron_pops.append(self.g_model.add_neuron_population(
-                    "if"+str(i-1),n_units[i-1],if_model,if_params,if_init)
+                    "if"+str(i-1),n_units[i-1],if_model,sparse_if_params,if_init)
                 )
             
-            # Postsynaptic neuron
-            self.neuron_pops.append(self.g_model.add_neuron_population(
-                "if"+str(i),n_units[i],if_model,if_params,if_init)
-            )
-
-            # Synapse
-            self.syn_pops.append(self.g_model.add_synapse_population(
-                "syn"+str(i-1)+str(i),"SPARSE_INDIVIDUALG",genn_wrapper.NO_DELAY,
-                self.neuron_pops[i-1],self.neuron_pops[i],
-                "StaticPulse",{},{'g':gw_vals[i-1]},{},{},
-                "DeltaCurr",{},{})
-            )
-
-            self.syn_pops[-1].set_sparse_connections(gw_inds[i-1][0],gw_inds[i-1][1])
+            if isinstance(relevant_layers[i-1],tf.keras.layers.Dense):
+                # Postsynaptic neuron
+                self.neuron_pops.append(self.g_model.add_neuron_population(
+                    "if"+str(i),n_units[i],if_model,dense_if_params,if_init)
+                )
+
+                # Synapse
+                self.syn_pops.append(self.g_model.add_synapse_population(
+                    "syn"+str(i-1)+str(i),"DENSE_INDIVIDUALG",genn_wrapper.NO_DELAY,
+                    self.neuron_pops[i-1],self.neuron_pops[i],
```

Comment:
you could use ``self.neuron_pops[-2], self.neuron_pops[-1]`` to make this clearer